NuGet.Services.Dashboard.Operations/Infrastructure/CommandLineParser.cs

```diff
@@ -70,7 +72,16 @@ public void ExtractOptions(ICommand command, IEnumerator<string> argsEnumerator)
                     throw new CommandLineException(TaskResources.MissingOptionValueError, option);
                 }
 
-                AssignValue(command, propInfo, option, value);
+                string newValue;
+
+                if (SecretConfigurationManager.TryInjectSecret(value, out newValue))
```

Comment:
>SecretConfigurationManager [](start = 20, length = 26)

this code is not object oriented. Please follow the ISecretReaderFactory pattern used in other area. 
CommandLineParser should get an ISecretReaderFactory parmeter in ctor. Look for examples in NuGet.Jobs or talk to me.